### PR TITLE
Improve styles for links containing icons (and related issues)

### PR DESCRIPTION
**Pull request by crhallberg**
TODO:
- [x] Author links may present a problem (discuss)
- [x] Titles of related works
- [x] Feedback link missing underlining
- [x] Delete links in search history need adjustment
- [x] View links above search results need adjustment
- [x] Checkbox facets need adjustment
- [x] Search tool links at bottom of results need adjustment
- [x] OpenURL links need adjustment
- [x] DOI links (doi.js) need adjustment 
- [x] Add to Book Bag on record page needs underlining removed
- [x] First/Prev/Next/Last buttons on record pages need review and to be made consistent (are they links or buttons?)
- [x] Place Title Level Request link needs adjustment
- [x] Show Other Versions (in results and on record view) needs adjustment
- [x] Inconsistent styles on request links (place hold underlined, storage retrieval / ILL not underlined)
- [x] All action links in Library Cards have also the icon underlined (Add Library Card, Edit, Delete)
- [x] Fix failing integration tests
- [x] Fix search history delete button
- [x] Fix advanced search link underlining
- [x] Fix delete button on "edit list item" page
- [x] Fix result-list.phtml full text link formatting
- [x] Fix wrapping of long status display in MyResearch menu
- [x] Extended holdings template needs icon alignment help
- [x] Close [VUFIND-1533](https://openlibraryfoundation.atlassian.net/browse/VUFIND-1533) when merging

### CODE REVIEW

**Comment by demiankatz** [comment]
@crhallberg, looks like you need to run a `lessToSass` to make the build happy. I've also requested a review from @EreMaijala, since I think this is of interest to him, though he has quite a backlog, so I don't expect to hear from him about this right away.

**Review by EreMaijala — CHANGES_REQUESTED** [review]
While the approach seems fine, this needs refinement and an "all or nothing" change. As far as I can see there are currently two problems:

1.) Positioning of the label is not quite the same as before. It's a bit low compared to the icon and there's not as much space between the icon and the text as before. Something like `vertical-align: middle` would look better to me, though admittedly this is a pretty minor thing.

2.) QRCode and cart links are now missing the underlining and have inconsistent spacing. This highlights the need for doing this with all of the elements. To illustrate:

Before:
before

After:
after

**Comment by demiankatz** [comment]
Thanks for the progress on this, @crhallberg. Since it seems there are still some significant portions that need to be done, plus some discussion that needs to happen, and because I don't think this is a critical bug, do you think it might be worth re-targeting this to dev-9.0? We're going to have to redo some of the work to account for the new icon system anyway, so maybe it's less effort to just do it once instead of solving some of these problems in two different ways.

I'm still happy to proceed with this for 8.1 if you think we can get it done in time, but I'm just offering this alternative if it seems more realistic/appropriate to you.

**Comment by crhallberg** [comment]
I'd be fine targeting this to 9.0 but I will also get the lead out this week. Are there any issues besides the ones @EreMaijala raised (fixed) or in the checklist at the top?

**Comment by demiankatz** [comment]
Thanks, @crhallberg -- I'm not aware of anything beyond @EreMaijala's review and the checkboxes above. Happy to discuss the author link issue if you want to elaborate on what needs discussion. If you can take care of this quickly, I'd be very happy to get it out of the way, but of the open 8.1 issues, it's probably the lowest priority.

**Comment by demiankatz** [comment]
@crhallberg, I've just merged the latest dev into this PR to bring things up to date.

I also notice that icons and text are now quite close together in related records. Here's the before:

image

and after:

image

Should we put some more right margin on the icons?

**Comment by demiankatz** [comment]
This looks even worse in the bootstrap3 theme, with inconsistent color, alignment and spacing... Here's the before:

image

...and the after:

image

I think that perhaps relates to @EreMaijala's earlier comment about icon alignment.

**Comment by demiankatz** [comment]
The related items spacing looks good to me in bootprint3, but now it almost looks like too much in bootstrap3/sandal. Also, the icon has still changed color (though maybe that's okay/intentional -- it doesn't really bother me). Here's another before and after:

image

image

If you think this is okay, I can be persuaded to live with it -- but it just seems a little off to me.

**Comment by demiankatz** [comment]
Regarding the result list options -- save/cart/QR code -- they now look absolutely perfect in bootstrap3... but maybe a little tight in bootprint3, and slightly misaligned in sandal. Not sure how much we want to be perfectionist about this, but seemed worth mentioning.

Here are before-and-afters for Sandal. As you can see, it looked wrong before, and it looks slightly less wrong (but in a different way) after. There's also a color change.

image

image

And here are before-and-afters for bootprint3:

image

image

Again, the new version feels better to me and fixes some inconsistent link underlining... but the icons feel very close to the text.

I should also note that all of my testing/screenshots are coming from Firefox, in case you're seeing different behavior in Chrome.

File: themes/bootstrap3/less/components/search.less
Comment (demiankatz) [inline themes/bootstrap3/less/components/search.less]: Do you really want to make the authors bold? It feels very heavy to me.

File: themes/bootstrap3/templates/record/cart-buttons.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/record/cart-buttons.phtml]: I believe that this template is shared between the results list and the record view, though I could be mistaken. Regardless of whether that detail is correct, I am seeing a problem on the record view when the book bag is enabled: when you mouse over buttons, the link underline disappears for every button except the cart button, where the underline either stays in place or simply changes its width (depending on which theme you are using).

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks for the progress, @crhallberg -- this is definitely feeling increasingly like an improvement over what came before. However, I found a couple more small issues and nitpicks (though some may be more critical than others).

File: themes/bootstrap3/templates/record/cart-buttons.phtml
Comment (crhallberg) [reply themes/bootstrap3/templates/record/cart-buttons.phtml]: Taken care of! 👍

File: themes/bootstrap3/less/components/search.less
Comment (crhallberg) [reply themes/bootstrap3/less/components/search.less]: Good catch! 👍

**Comment by demiankatz** [comment]
@crhallberg, recent changes to release-8.1 introduced a conflict here; I went ahead and resolved it and tested that it didn't cause problems.

It does appear that the fix you made to the cart buttons on the record page caused the buttons to now display inconsistently/incorrectly in the search results. I'm now seeing things like this:

image

I'm increasingly feeling that it may be worthwhile to re-target this to the dev branch and release 9.0 so we have more time to work through these issues. I don't think this is a critical bug by any measure, so we can afford to take a little more time... and since the changes here are going to create conflicts with the icon work in dev, going ahead and re-targeting now means that we don't have to solve the problem twice in two different ways. What do you think? (For what it's worth, @EreMaijala also supports waiting on this -- he's on vacation until August, but I checked in for his opinion on this PR before he left).

**Comment by demiankatz** [comment]
@crhallberg, as we discussed last week, I have changed this to point at dev instead of release-8.1, as it seems safest to defer this work until the 9.0 release. Could you resolve conflicts when time permits?

**Comment by sturkel89** [comment]
@crhallberg - I reviewed things today and see lots of improvement! Here are some minor observations to add to the mix. I'll make one comment for each area of the site where I see a change that might need a tiny bit more attention.

**Cart options on search results pages:**

- Alignment is now great across the board. 
- Spacing between icon and text is great in sandal and bootstrap3; spacing is still tight in bootprint3 and could be tweaked to look more like the other two themes.
- Underlining of links is turned off by default for all three themes, while links are underlined on mouseover. (I'm not sure if we have any concerns about links not looking like links?)
- I notice that link color changes slightly on mouseover with all three themes, but in bootstrap3 the icon's color changes slightly on mouseover as well. I think it would be better if icon color were not affected by mouseover, as in the other two themes.

Before (bootstrap3):
image
After (bootstrap3):
image
Before (bootprint3):
image
After (bootprint3):
image

**Comment by sturkel89** [comment]
@crhallberg - more comments.

**Similar items** (within an item record)

- Spacing is much improved in bootstrap3 and sandal. 
- Spacing between icon and text in bootprint3 is still tight and should be tweaked to look more like the other two themes.

Before (sandal):
image
After (sandal):
image
Before (bootprint3):
image
After (bootprint3):
image

**Comment by sturkel89** [comment]
@crhallberg - last one.

**My Account** - Edit and Delete options for items on saved item list

- In all three themes, Edit and Delete were previously underlined links; now, all three themes have these links as plain blue text, but they do NOT underline on mouseover. I think they probably should underline on mouseover (or maybe be underlined all the time).
- In sandal and bootstrap3, the link text now changes color subtly upon mouseover.
- In bootprint3, there's more obvious highlighting of link text on mouseover.
- Spacing is good in sandal and bootstrap3 (before and after) but spacing is a bit tight in bootprint3 (both before and after). Can you improve spacing between icon and text in bootprint3?

Sandal (before):
image
Sandal (after):
image
Bootprint3 (before):
image
Bootprint3 (after):
image

File: themes/bootstrap3/templates/Related/Similar.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/Related/Similar.phtml]: @crhallberg, when you moved the definition of `$format` down lower, a bug was introduced here, since `$format` is not yet defined for this line. This was causing error screens when @sturkel89 looked at it in development mode.

**Comment by sturkel89** [comment]
@crhallberg, I see improvements across the board - but still some issues with link behavior on mouseover.

**QR Code/Bookbag/Save to List in Results browse**

- All three links are now underlined by default in all three themes -- looks great!
- On mouseover, **underlining disappears** in all three themes. This behavior differs from the other links on the page, all of which retain underlining on mouseover. For consistency, I think underlining should be brought back.
- In Sandal and Bootstrap3, link text changes color on mouseover. The color change is consistent with behavior of other links on the page; no adjustment needed to colors.
     
- In Bootprint3, this link text gets darker on mouseover, while other link text on the page changes to bright blue on mouseover. For consistency, mouseover color here should probably match mouseover color in other link text on the page. (I'm not sure which color change is better, the bright blue or the darker color!)

Bootprint example: 
image

**Logged-in areas**

- In the Your Account menu, each link is underlined in every theme. Link text underlining disappears on mouseover; I think this is new behavior.
- In a list of Saved Items, the Edit/Delete links are now underlined. The underline disappears on mouseover.
-  Color issues mentioned above are the same here; link text color change on mouseover is fine in Sandal and Bootstrap, but in Bootprint it diverges from rest of links on page. 

**Options at the top of item records**

- Not sure how these worked before, but the Cite This, Text This, etc. buttons across the top of every item record have non-underlined link text by default (except for the Book Bag button which is underlined by default) -- but upon mouseover, the underline appears. (Background color also changes, so underlining seems unnecessary in these buttons.)
- Most of these mouseover underlines show underlining in the gap between icon and link text -- in every case except the Book Bag button.
- This is consistent across the three themes.
- Ideal fix would be to do away with all underlining in these buttons. This is how the Email/Delete/Export/Print/Book Bag buttons work at the top of a Saved Items list - no underlining ever, and button background changes color on mouseover.
- (The Book Bag icon in Bootstrap and Sandal is also a different color from the other icons in this area.)

Sandal:
image

Bootprint3:
image

Bootstrap3:
image

**Comment by sturkel89** [comment]
@crhallberg - Minor spacing issue remains in Similar Items list in Bootprint. 

Before, spacing between icon and text was REALLY tight:
image

Now, it's better, but still a bit tight:
image

Spacing is much more open in the other two themes - example from Sandal:
image

**Comment by demiankatz** [comment]
@crhallberg, I just merged this with dev and recompiled the CSS so that CI could proceed. However, I happened to notice an issue in the process -- the Sandal theme now shows nearly-invisible blue icons (as opposed to the white icons visible in the dev branch) in the header. Screenshot:

image

**Comment by demiankatz** [comment]
@crhallberg, after merging icon helper functionality from #1962 into dev, there are some non-optimal-looking links in some situations...

Bootprint3 theme:

image

Note how tight the icons/text are, and also the underlined space at the end of "Place a Hold."

Sandal theme:

image

...some very unfortunate wrapping there.

Is this a good place to address these issues?

**Comment by crhallberg** [comment]
Doing my best to address issues while making sure the merge with #1962 goes smoothly and with little redundancy.

File: themes/root/css/icon-helper.css
Comment (demiankatz) [inline themes/root/css/icon-helper.css]: Since this entire file has been removed from #1962, does this line need to be translated somewhere over there? Just mentioning it so we don't lose track of it.

**Review by demiankatz — COMMENTED** [review]
See one small comment below.

Also, we seem to have a problem now where on the record page, the "Add to Book Bag" button has underlined text when nothing else does:

image

...I think I see this problem in some of @sturkel89's screen shots above, though it wasn't specifically highlighted there. I think the key thing is that we want to fix the record view without breaking the underlining in the search results view. Tricky!

**Comment by crhallberg** [comment]
Just merged and the merge and a typo fix took care of the Bookbag button and hold link problems!

**Comment by demiankatz** [comment]
@crhallberg, @EreMaijala did some testing of this this morning and found some further issues that need adjusting. I've added TODO checkboxes for all of his observations. I'll see if I can find time to try to fix some, and I'll let you know what I need help with.

**Comment by demiankatz** [comment]
@crhallberg, it also looks like the integration tests are failing -- presumably some selectors need to be adjusted:

```
There were 2 failures:

1) VuFindTest\Mink\FavoritesTest::testAddSearchItemToFavoritesNewAccount
Element not found: .save-record index 0
Failed asserting that false is true.

/usr/local/vufind/module/VuFind/src/VuFindTest/Integration/MinkTestCase.php:365
/usr/local/vufind/module/VuFind/src/VuFindTest/Integration/MinkTestCase.php:452
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/FavoritesTest.php:252
/usr/local/vufind/module/VuFind/src/VuFindTest/Feature/AutoRetryTrait.php:99
phpvfscomposer:///usr/local/vufind/vendor/phpunit/phpunit/phpunit:97

2) VuFindTest\Mink\HomePageFacetsTest::testHierarchicalFacets
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'level1a level1z'
+''

/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/HomePageFacetsTest.php:69
/usr/local/vufind/module/VuFind/src/VuFindTest/Feature/AutoRetryTrait.php:99
phpvfscomposer:///usr/local/vufind/vendor/phpunit/phpunit/phpunit:97
```

Given that a bit more work is still needed to finish this PR, I'm going to refocus on trying to finish up some other things before I do more work here. We can touch base once you're online to figure out next steps.

**Comment by crhallberg** [comment]
Thank you for the detailed feedback, @EreMaijala! I'll get to work!

**Comment by demiankatz** [comment]
Thanks for the progress on this, @crhallberg! I retried the two tests that I noted as failing in my comment from last week. The HomePageFacetsTest is passing now, but FavoritesTest is still failing with the same error. Can you investigate? I've added a TODO checkbox about this.

I'll also try to do some hands-on testing when time permits, and I'll let you know if I encounter any issues.

**Comment by demiankatz** [comment]
@crhallberg, I started looking at this today, and it's a great improvement so far, except for one thing -- the icon links seem to be a little higher than they should be, as if they're getting pushed up with a bottom margin or something. It's especially pronounced in the Sandal theme, but visible everywhere. You can notice it in the search tools, for example, where the links are out of line with the heading and the divider dashes.

Before:

image

After:

image

Does this have an easy fix?

**Comment by crhallberg** [comment]
@demiankatz there was definitely a lost class in the HTML that caused the first failure. Not sure why the hierarchical facets are failing.

**Comment by demiankatz** [comment]
Thanks, @crhallberg. The hierarchical facet failure seemed to go away on its own, so hopefully everything is addressed now. I'll re-run the test suite and look things over as soon as time permits (probably early next week). I'm also interested to hear whether @EreMaijala has any feedback regarding the OpenURL link, since I think that's the last thing we weren't 100% sure about.

**Comment by demiankatz** [comment]
I can confirm: all tests are passing now.

**Comment by demiankatz** [comment]
One minor observation, @crhallberg: the underline seems to have disappeared from the "Advanced" link next to the search box. Since this actually is a link, and not a button, shouldn't the underline remain?

File: themes/bootstrap3/templates/search/history-table.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/search/history-table.phtml:63]: Looks like you fixed the Save icon here, but not the Delete icon above; see:

image

File: themes/bootstrap3/templates/RecordDriver/DefaultRecord/list-entry.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/RecordDriver/DefaultRecord/list-entry.phtml:193]: This link looks good, but if you click on it, the destination page has a "delete" link that needs to be fixed up:
image

File: themes/bootstrap3/templates/RecordDriver/DefaultRecord/result-list.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/RecordDriver/DefaultRecord/result-list.phtml]: Does the link text need a span around it? The underline looks a little too big here:
image
Maybe this is also an opportunity to put text and icon on separate lines in the template to improve readability.

File: themes/bootstrap3/templates/myresearch/menu.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/myresearch/menu.phtml]: Something weird is going on with wrapping when there are a lot of statuses. See, for example:

image

(compared to the current dev branch:
image
)

This is a little bit hard to reproduce. Here's my advice:

1.) Turn on the Demo driver
2.) Go to the profile page and log in (make up whatever credentials you like) to activate patron functionality
3.) Keep logging in and out until you get three boxes next to Checked Out Items. Since things are re-randomized every time, it will take several attempts. To improve the odds, you can bump up the transaction count by modifying the $trans variable at the top of getRandomTransactionList in the demo driver.

I don't think it matters whether or not you have historic transactions turned on, but if you need them for some reason, they're off by default but can be activated in Demo.ini's [TransactionHistory] section.

**Comment by demiankatz** [comment]
@crhallberg, I haven't had a chance to re-test everything you've changed, but I've found several new issues, so you still have some things to work on. I'll resume testing after you've had time to make some more fixes. Please let me know when you're ready!

File: themes/bootstrap3/templates/myresearch/menu.phtml
Comment (crhallberg) [reply themes/bootstrap3/templates/myresearch/menu.phtml]: Another easy way to test this is to edit the session storage in the browser to something like `{ "ok": 10, "warn": 1, "overdue": 5 }`.

Comment (demiankatz) [reply themes/bootstrap3/templates/myresearch/menu.phtml]: Yes, that's much easier -- thanks!

**Comment by demiankatz** [comment]
@crhallberg, something has caused the "inconsistent request link underlining" problem to come back:

image

I've unchecked the associated checkbox until we can re-fix it.

**Comment by demiankatz** [comment]
@crhallberg, it also looks like the search history delete button (to remove a search after you have saved it) still needs tweaking:

image

Additionally, while the MyResearch menu wrapping problem is fixed, the status icons used to be right-aligned, and now they aren't any more. I'm not sure if this is a problem, but it does look a little less tidy than before.

Before:

image

After:

image

**Comment by demiankatz** [comment]
@crhallberg, thanks for the progress on this. I noticed that the "inconsistent underlining on request links" problem was not fixed, so I did a little digging, and I think I took care of it with [8b2f914](https://github.com/vufind-org/vufind/pull/2400/commits/8b2f914593ad53bdaf7c46b9c862f89103635238) -- could you please review my changes? Also, I resolved merge conflicts and updated this with the latest dev branch.

**Comment by demiankatz** [comment]
@crhallberg, the next detail that could probably use some attention -- if you turn on extended holdings mode by setting holdingsTemplate to "extended" in config.ini, the request links look pretty weird now (though they didn't look great to begin with).

Before:

image

After:

image

Any ideas? I think at least top-aligning the icon might help.

File: themes/bootstrap3/templates/ajax/resolverLink.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/ajax/resolverLink.phtml]: @crhallberg, does this change serve any purpose? It looks like the markup is the same, but it's been made less readable by being mashed into a single line. Was the extra whitespace causing a problem that needed to be addressed, or was this an accidental change? If it's intentional, we should add a comment explaining why so that it doesn't get reformatted in the future. If it's not intentional, we should revert it to eliminate an unnecessary change.

**Review by demiankatz — CHANGES_REQUESTED** [review]
@crhallberg, one more question (see below).

I haven't tested absolutely everything -- I'm saving the hardest parts to test (Overdrive, etc.) for last -- but once the last few open issues here are sorted out, I'll take a closer look at those. We're nearly done!

**Comment by EreMaijala** [comment]
I pushed a fix for DOI links. Includes a style fix for any other icon-link in the record metadata array.

**Comment by demiankatz** [comment]
@crhallberg, I'm seeing test failures in this branch right now:

```
There were 5 failures:

1) VuFindTest\Mink\AccountMenuTest::testIconNone
Element not found: #account-icon .account-status-none index 0
Failed asserting that false is true.

/usr/local/vufind/module/VuFind/src/VuFindTest/Integration/MinkTestCase.php:365
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php:320
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php:345
/usr/local/vufind/module/VuFind/src/VuFindTest/Feature/AutoRetryTrait.php:99
phpvfscomposer:///usr/local/vufind/vendor/phpunit/phpunit/phpunit:97

2) VuFindTest\Mink\AccountMenuTest::testIconGood
Element not found: #account-icon .account-status-good index 0
Failed asserting that false is true.

/usr/local/vufind/module/VuFind/src/VuFindTest/Integration/MinkTestCase.php:365
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php:320
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php:364
/usr/local/vufind/module/VuFind/src/VuFindTest/Feature/AutoRetryTrait.php:99
phpvfscomposer:///usr/local/vufind/vendor/phpunit/phpunit/phpunit:97

3) VuFindTest\Mink\AccountMenuTest::testIconWarning
Element not found: #account-icon .account-status-warning index 0
Failed asserting that false is true.

/usr/local/vufind/module/VuFind/src/VuFindTest/Integration/MinkTestCase.php:365
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php:320
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php:379
/usr/local/vufind/module/VuFind/src/VuFindTest/Feature/AutoRetryTrait.php:99
phpvfscomposer:///usr/local/vufind/vendor/phpunit/phpunit/phpunit:97

4) VuFindTest\Mink\AccountMenuTest::testIconDanger
Element not found: #account-icon .account-status-danger index 0
Failed asserting that false is true.

/usr/local/vufind/module/VuFind/src/VuFindTest/Integration/MinkTestCase.php:365
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php:320
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php:396
/usr/local/vufind/module/VuFind/src/VuFindTest/Feature/AutoRetryTrait.php:99
phpvfscomposer:///usr/local/vufind/vendor/phpunit/phpunit/phpunit:97

5) VuFindTest\Mink\AccountMenuTest::testIconClashes
Element not found: #account-icon .account-status-danger index 0
Failed asserting that false is true.

/usr/local/vufind/module/VuFind/src/VuFindTest/Integration/MinkTestCase.php:365
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php:320
/usr/local/vufind/module/VuFind/tests/integration-tests/src/VuFindTest/Mink/AccountMenuTest.php:412
/usr/local/vufind/module/VuFind/src/VuFindTest/Feature/AutoRetryTrait.php:99
phpvfscomposer:///usr/local/vufind/vendor/phpunit/phpunit/phpunit:97

FAILURES!
Tests: 2228, Assertions: 11268, Failures: 5.
```

I've run AccountMenuTest in the dev branch and confirmed that all tests are passing there. Can you investigate what is going wrong here?

**Comment by crhallberg** [comment]
I found a frustratingly simple fix for this: I accidentally merged the ids and classes on the account ajax icon. I think that will fix the errors above, if you have time to run the integration suite again.

**Comment by demiankatz** [comment]
Thanks, @crhallberg, tests are passing again. I'll do some more manual hands-on testing now and let you know if I find new issues.

**Comment by demiankatz** [comment]
@EreMaijala / @crhallberg, I'm trying out the DOI icon fixes now that the DOI Demo driver is available to make this task easier... and I notice a bit of an odd alignment issue.

Before (dev branch):

image

After (this PR):

image

Do you think we should top-align the icon to make this look a little tidier?

**Comment by demiankatz** [comment]
@crhallberg, I'm seeing a similar issue in the extended holdingsils output:

Before (dev branch):

image

After (this PR):

image

Once again, I feel that a top vertical alignment would make this look a lot neater. Hopefully a single fix can address both this and the previously-mentioned problem.

**Comment by demiankatz** [comment]
@crhallberg, this might just be a matter of taste, but is the down arrow a little too close to the text in the header drop-downs now?

Before (dev branch):

image

After (this PR):

image

What do you think?

**Review by demiankatz — CHANGES_REQUESTED** [review]
@crhallberg, please see the above three comments for a few issues that might need attention. I think we're very close to done -- I've tested everything changed here, except for the Overdrive code. I'll do a round of Overdrive testing (as best I can, by providing fake data in the controller) once everything else is polished.

**Comment by EreMaijala** [comment]
> @EreMaijala / @crhallberg, I'm trying out the DOI icon fixes now that the DOI Demo driver is available to make this task easier... and I notice a bit of an odd alignment issue.
> 
> Before (dev branch):
> 
> image
> 
> After (this PR):
> 
> image
> 
> Do you think we should top-align the icon to make this look a little tidier?

Yes, the dev branch looks cleaner to me. Well, in this case I'd actually prefer a mix of both so that the text lines would be left-aligned like in this PR, but with a top-aligned icon. This might not be universally the same particularly if there are entries with and without icons, though.

**Comment by crhallberg** [comment]
I've aligned the icons to the top for multi-line and I've added a gap to the dropdowns.

**Comment by demiankatz** [comment]
Thanks, @crhallberg -- all the latest fixes look good to me. I've also resolved a conflict with dev. I think all that's left is testing the Overdrive-related templates, which I'll try to get to later this week!

**Comment by demiankatz** [comment]
@crhallberg, with the Overdrive testing now possible, I think I've gotten through all the testing I possibly can. I've found just one last issue to ask about:

I notice that the vertical alignment of the badge in the facet results has moved from top to center...

Before:

image

After:

image

Was this intentional? I honestly don't have strong opinions about whether one is better than the other, but I happened to notice it and thought it worth checking about.

I'm happy to approve this as-is, but let me know if you think further change is desirable. I'm also going to ask @EreMaijala to take one last look at this point in case he has any further observations to share.

**Comment by demiankatz** [comment]
Also, just for the record: I merged the latest dev into this, resolved conflicts, re-ran tests (all still passing!) and renamed the PR to something that I think better reflects what it actually does (since scope expanded after the start of the project).

**Review by EreMaijala — CHANGES_REQUESTED** [review]
Looks pretty good. Advanced search has a misaligned plus button "Add Search Field":
Advanced Search

Also a couple of other places could be tweaked a bit, but they're no worse than before:
Underlined buttons in Profile page:
Profile Page

Cart looks pretty bad:
Cart

**Review by demiankatz — APPROVED** [review]
Thanks for catching these issues, @EreMaijala, and for fixing them, @crhallberg. At this point, it looks like all known problems are solved, and the full test suite is still passing. I think I can merge this once @EreMaijala provides final approval.

**Review by EreMaijala — APPROVED** [review]
Approved! :)

**Comment by demiankatz** [comment]
Thanks, @EreMaijala and @crhallberg -- a monumental effort! :-)

**Pull request merged**
